analytics/lib/counts.py

```diff
@@ -425,11 +479,17 @@ def do_pull_minutes_active(property: str, start_time: datetime, end_time: dateti
         zerver_userprofile.id = zerver_useractivityinterval.user_profile_id
     WHERE
         zerver_useractivityinterval.end >= %%(time_start)s AND
+        {realm_clause}
         zerver_useractivityinterval.start < %%(time_end)s
     GROUP BY zerver_userprofile.id %(group_by_clause)s
-"""
+""".format(realm_clause=realm_clause)
 
-count_realm_active_humans_query = """
+def count_realm_active_humans_query(realm: Optional[Realm]) -> str:
+    if realm is None:
+        realm_clause = ""
+    else:
+        realm_clause = "realm_id = %%(realm_id)s AND"
+    return """
```

Comment:
One other question I had is how much you've thought about the API detail of whether we have the `_query` functions insert the `realm_id` themselves or just have `realm_clause = "realm_id = %(realm_id)s AND" % (realm.id,)` in this case?  I think the latter would result in cleaner code (since we already passed the `Realm` object here, there's no reason to not do the substitution now before we pass it into SQL).

Reply:
@timabbott I think  realm_clause = "realm_id = %(realm_id)s AND" % (realm.id,) would be better as it will result in cleaner code as well it makes us bound to pass realm in the other direction too i.e other than COUNT_STATS we have to pass realm in do_fill_count_stat_at_hour which I suppose is a good practice as we realm is required there. Though I am confused about whether to pass realm_id with string replacement or by giving it as a parameter to cursor.execute()

Reply:
Yeah, I'd appreciate it if you made that change.  I think either `count_message_type_by_user_query` should take a boolean (for whether there's a realm) or it should take an `Optional[realm]` and do the realm-level filtering itself (rather than expecting the resulting query to have a realm passed in).  

The latter option (i.e. what I proposed above) seems cleaner to me; can you make that change?